remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/indexing/ProtobufEntityConverter.java

```diff
@@ -0,0 +1,80 @@
+package org.infinispan.query.remote.impl.indexing;
+
+import java.io.IOException;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.infinispan.commons.CacheException;
+import org.infinispan.protostream.ProtobufParser;
+import org.infinispan.protostream.SerializationContext;
+import org.infinispan.protostream.WrappedMessage;
+import org.infinispan.protostream.descriptors.Descriptor;
+import org.infinispan.query.remote.impl.mapping.reference.GlobalReferenceHolder;
+import org.infinispan.search.mapper.mapping.EntityConverter;
+
+public class ProtobufEntityConverter implements EntityConverter {
+
+   public static final String SCALAR_VALUE_ENTITY_NAME = "$$ScalarEntity$$";
+
+   private final SerializationContext serializationContext;
+   private final Set<String> indexedMessageTypes;
+
+   public ProtobufEntityConverter(SerializationContext serializationContext, Set<GlobalReferenceHolder.RootMessageInfo> rootMessages) {
+      this.serializationContext = serializationContext;
+      this.indexedMessageTypes = rootMessages.stream().map(rootMessage -> rootMessage.getFullName()).collect(Collectors.toSet());
+   }
+
+   @Override
+   public boolean appliesTo(Object entity) {
+      return entity instanceof ProtobufValueWrapper;
+   }
+
+   @Override
+   public ConvertedEntity convert(Object entity) {
+      if (!appliesTo(entity)) {
```

Comment:
I may be wrong, but I think this check is unnecessary as callers always call `appliesTo` before calling this.

Reply:
right I don't like defensive programming either… removing the check